### ISSUE TITLE
Multiple user sessions create interference

### ISSUE THREAD

**Issue by abollini** [DSpace/DSpace#3249]
**Describe the bug**
When the same user logged-in multiple times in parallel there are side effects on the other login after the first logout.
The first logout indeed disconnect all the session as they are verified against a single sessionSalt stored at the EPerson level.

**To Reproduce**
1. login in chrome as dspacedemo+admin
2. start a new submission 
2. login in firefox as  dspacedemo+admin
3. logout from firefox
4. try to add a metadata in the chrome browser, you will get an error on the server side (a 403 response). The angular UI is loss no way to solve other than an hard refresh
5. the same would happen if you click on the bitstream registry, add a new entry and try to save... in general almost any edit feature on the angular side is broken

**Expected behavior**
1. If I log out from firefox I expect to stay logged-in in chrome, the same is not true for multiple tab on a single browser in such case I would expect that the logout from any tab is propagated to all the other tabs (i.e. there is only a "login session" shared between all the tab).
2. when this event happen I would like ideally to have angular to automatically detect the issue before hitting an error but at least angular should redirect me to the login once that the issue occurs.

Please note that point 2 can be also a partial temporary solution for the multi-browsers issue. Keep also in mind that the multi-browsers issue can be hit also by different users sharing the same account (as typically done during the testhaton)

**Related work**
The are two main problems on the rest side
1. when a request get a JWT token related to a session salt that has been invalidated the token is discarded and a 403 is reported back instead than a 401 that seems more appropriate as indeed there are no user authenticated. It would be hard to distinguish between the scenario where the JWT token is really invalid (corrupted/hijacked), expired or invalidated on the server side (session salt). So I suggest to simply threat all of them with a 401
2. there is only a session salt and the second browser (or the second user using the same account) currently reuse the same salt, this mean that when it logged-out invalidate also the other user/browser session. We should have multiple valid session salt for each user so that many simultaneous sessions can be managed. Each browser should uniquely identify itself so that multiple tabs in the same browser can reuse the same session salt (this could be unnecessary as right now thanks to the fact that the jwt token is stored in a cookie the new tabs are automatically loggedin)

Just to anticipate, the activity 1 requires only minor changes on the angular side as the 401 is already managed with a redirection to the login. The only change needed seems to be related to a second check that pretend an eventual jwt token to be expired (this is not the case when the logout is triggered by a different user, browser or tab).
The activity 2 would required the angular side to create a random identifier and store it in a browser cookie if one doesn't yet exists so that other tab will reuse it.

**Comment by abollini** [comment]
Activity 1 should require 2 hours on the REST side with the simple fix (return 401)

Activity 2 is much more complex and involve also a db changes, I estimate 3 days of work on the REST side

**Comment by tdonohue** [comment]
@abollini : I have a feeling we should split these activities apart into two tickets.  I agree that Activity 1 might be good to fix for 7.0...it gives a more appropriate 401 response code.

However, Activity 2 seems like a scenario that is **unlikely to occur frequently in Production** (correct me if I'm wrong).   So, we could think to document it as a known issue in 7.0, giving us more time to plan out a solution in 7.1 or similar.

**Comment by tdonohue** [comment]
As promised in today's meeting, I've split this into two tickets:
* https://github.com/DSpace/DSpace/issues/3256 - The high priority task to return a 401 when logout occurs.
* https://github.com/DSpace/DSpace/issues/3257 - The low priority task to support multiple active sessions per user.

Therefore, I'm closing this ticket as all further discussion should occur on one of the tickets above.
